org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/AbstractJavaModelTests.java

```diff
@@ -369,8 +369,11 @@ public synchronized void elementChanged(ElementChangedEvent event) {
 				System.arraycopy(this.deltas, 0, copy, 0, this.deltas.length);
 				copy[this.deltas.length]= event.getDelta();
 				this.deltas= copy;
-
-				new Throwable("Caller of IElementChangedListener#elementChanged with delta " + event.getDelta()).printStackTrace(new PrintStream(this.stackTraces));
+				StringBuilder message = new StringBuilder();
+				message.append('[').append(Thread.currentThread().getName());
+				message.append("] Caller of IElementChangedListener#elementChanged with delta ");
```

Comment:
One could add Job.currentJob() too?

Reply:
> One could add Job.currentJob() too?

Sounds good, thanks.

_(it seems the idiom is `Job.getJobManager().currentJob()`)_